independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/logging/DefaultClientLogger.java

```diff
@@ -50,8 +62,19 @@ private String bodyToString(Buffer body) {
     }
 
     private String asString(MultiMap headers) {
-        return headers.entries().stream()
-                .map(e -> String.format("%s=%s", e.getKey(), e.getValue()))
-                .collect(Collectors.joining(" "));
+        if (headers.isEmpty()) {
+            return "";
+        }
+        StringBuilder sb = new StringBuilder((headers.size() * (6 + 1 + 6)) + (headers.size() - 1)); // this is a very rough estimate of a result like 'key1=value1 key2=value2'
```

Comment:
I fail to see why we should make this more complicated if only used for debug logging? I can understand doing things like that in a hot path but here?

Reply:
Complicated how? It's just iterating over the headers and building up a String. How is that more complicated than was was being done with lambdas?

Reply:
14 lines vs 3 lines, pre-calculating the header size, iterating the entries, checking if the entry is the first, and so on... yes this has added complexity.

Sure, the streams, lambda, and String.format approach will be slower, but it's more readable, and if it's already guarded by `isDebugEnabled()` and is not in a hot path, then this might not add value in the final execution.

Having said that I appreciate that Quarkus tries to squash every ms of performance, but it might be better to focus on more critical paths.

Reply:
> 14 lines vs 3 lines, pre-calculating the header size, iterating the entries, checking if the entry is the first, and so on... yes this has added complexity.

No it's not

Reply:
> Having said that I appreciate that Quarkus tries to squash every ms of performance, but it might be better to focus on more critical paths

I appreciate the input, but the squezing performance out of Quarkus is an ongoing task and in this case I happen to come acroos something that was not performant and I addressed it, it literally took 5 minutes, it's not like it took time from other tasks...

It's always good to keep in mind the context of how development works in a large project when attempting to apply generalized statements.

Reply:
> I truly appreciate the input, but the squezing of performance out of Quarkus is an ongoing task. In this specific case I just happen to come across something that was not performant (this is important as I did not go hunting for performance issues and come up with this...) and I addressed it - it literally took 5 minutes, it's not like it took time from other tasks...
> 
> It's always good to keep in mind the context of how development works in a large project when attempting to apply generalized (but otherwise well intended) statements.

Thank you, I know this is a different kind of project and I didn't want to apply generalized statements, I was following the same observation of Guillaume, I can read the code perfectly (and probably everyone involved in the project can), but claiming that anyone that knows even the slightest Java can write without any effort, well, it's just not realistic.

Anyway, since the context of the project is different, it's perfectly fine. 👍🏽